src/main/java/io/rcktapp/api/handler/firehose/FirehoseDb.java

```diff
@@ -45,36 +45,36 @@ public class FirehoseDb extends Db
      */
     protected String includeStreams;
 
-    AmazonKinesisFirehoseAsync firehoseClient = null;
+    protected AmazonKinesisFirehoseAsync firehoseClient = null;
 
     public FirehoseDb() {
         super();
         setType("firehose");
     }
 
     @Override
-    public void bootstrapApi() throws Exception {
-        List<Pair<String, String>> nameActuals = new ArrayList<>();
+    public void bootstrapApi() {
         // our local streams
-        getFirehoseClient().listDeliveryStreams(new ListDeliveryStreamsRequest().withDeliveryStreamType("DirectPut")).getDeliveryStreamNames().forEach(name -> nameActuals.add(Pair.of(name.toLowerCase(), name)));
+        Stream<Pair<String, String>> inKinesis = firehoseNameStream().map(name -> Pair.of(name.toLowerCase(), name));
```

Comment:
And just like that, all of the paging is transparent and not a concern for this class at all!  We just get a virtually infinitely long available stream of strings we can map.